samples/getvmsbycluster.py

```diff
@@ -0,0 +1,182 @@
+                        help='Password to use when connecting to host')
+    parser.add_argument('--csv', required=False, action='store_true',
+                        help='Write output to csv file')
+    parser.add_argument('--csvfile', required=False, action='store',
+                        default='getvmsbycluster.csv',
+                        help='Filename and path of csv file')
+    parser.add_argument('--json', required=False, action='store_true',
+                        help='Write out to json file')
+    parser.add_argument('--jsonfile', required=False, action='store',
+                        default='getvmsbycluster.json',
+                        help='Filename and path of json file')
+    parser.add_argument('--silent', required=False, action='store_true',
+                        help='supress output to screen')
+    args = parser.parse_args()
+    return args
+
+
+def vmsummary(summary):
+    vmsum = {}
+    config = summary.config
+    ipaddr = summary.guest.ipAddress
+    vmsum['mem'] = str(config.memorySizeMB / 1024)
+    vmsum['disk'] = str(summary.storage.committed / 1073741824)
+    vmsum['cpu'] = str(config.numCpu)
+    vmsum['path'] = config.vmPathName
+    vmsum['guestname'] = config.guestFullName
+    vmsum['state'] = summary.runtime.powerState
+    vmsum['managedby'] = config.managedBy if config.managedBy else ''
+    vmsum['annotation'] = config.annotation if config.annotation else ''
+    vmsum['ip'] = ipaddr if ipaddr is not None else ''
+
+    return vmsum
+
+
+def vmprint(dc, cluster, host, vm, summary):
+    print("VM: " + vm.summary.config.name + " Host: " + host, end="")
+    print(" Folder: " + vm.parent.name + " Cluster: " + cluster, end="")
+    print("    IP: " + summary['ip'] + " CPU: " + summary['cpu'], end="")
+    print(" Mem: " + summary['mem'] + " Disk: " + summary['disk'])
+    print("    State: " + summary['state'], end=""),
+    print(" Managedby: " + summary['managedby'], end=""),
+    print(" Path: " + summary['path'])
+    if summary['annotation'] is not '':
+        print("    Annotation: " + summary['annotation'])
+
+
+def vm2dict(dc, cluster, host, vm, summary):
+    # If nested folder path is required, split into a separate function
+    vmname = vm.summary.config.name
+    data[dc][cluster][host][vmname]['folder'] = vm.parent.name
+    data[dc][cluster][host][vmname]['mem'] = summary['mem']
+    data[dc][cluster][host][vmname]['disk'] = summary['disk']
+    data[dc][cluster][host][vmname]['cpu'] = summary['cpu']
+    data[dc][cluster][host][vmname]['path'] = summary['path']
+    data[dc][cluster][host][vmname]['ip'] = summary['ip']
+    data[dc][cluster][host][vmname]['guestname'] = summary['guestname']
+    data[dc][cluster][host][vmname]['state'] = summary['state']
+    data[dc][cluster][host][vmname]['managedby'] = summary['managedby']
+    data[dc][cluster][host][vmname]['annotation'] = summary['annotation']
+
+
+def data2csv(data, args):
```

Comment:
After applying this to my actual use case I definitely plan to take out the csv export so I can list multiple IPs per VM. Also I was thinking about changing the output to a json pretty print if that would be agreeable.  I'd like to leave in the 3-4 lines required to support saving to json, but I'll take it out if you want me to.